### PR TITLE
Generate nested analysis results: features are contained by their namespaces

### PR DESCRIPTION
**Pull request by justinfagnani**
- [ ] CHANGELOG.md has been updated

### CODE REVIEW

File: src/test/static/analysis/namespaces/analysis.json [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+        "start": {
+          "column": 2,
+          "line": 7
+        }
+      },
+      "summary": "This is the \"aaa\" namespace. (summary)",
+      "functions": [
+        {
+          "description": "This is a simple namespace method.",
+          "name": "aaa.sum",
```
Comment (rictic) [inline src/test/static/analysis/namespaces/analysis.json]: Hm, is this better as `sum` or `aaa.sum`?

File: src/analysis-format.ts [hunk 2]
```diff
@@ -12,15 +12,24 @@
  * http://polymer.github.io/PATENTS.txt
  */
 
-export interface Elements {
+/**
+ * The global namespace of features contained in an analysis.
+ *
+ * Top-level members are unnamespaced, evenrything else is contained
```
Comment (justinfagnani) [inline src/analysis-format.ts]: done

File: src/generate-analysis.ts [hunk 3]
```diff
@@ -115,7 +149,7 @@ export class ValidationError extends Error {
  * Throws if the given object isn't a valid AnalyzedPackage according to
  * the JSON schema.
  */
-export function validateElements(analyzedPackage: Elements|null|undefined) {
+export function validateElements(analyzedPackage: Analysis|null|undefined) {
```
Comment (justinfagnani) [inline src/generate-analysis.ts]: done

File: src/test/static/analysis/namespaces/analysis.json (hunk 1 above)
Comment (justinfagnani) [inline src/test/static/analysis/namespaces/analysis.json]: Names include the namespace. We may want to change this in the (near) future, but that's how it is now.

**Pull request merged**